pkg/kubectl/cmd/util/factory.go

```diff
@@ -303,6 +307,34 @@ func (f *Factory) BindFlags(flags *pflag.FlagSet) {
 	flags.SetNormalizeFunc(util.WordSepNormalizeFunc)
 }
 
+func (f *Factory) AddDeprecatedFlagShorthands(shorthand string) error {
```

Comment:
Thanks @deads2k @eparis, I'll create a separate PR in github.com/spf13/pflag to support deprecating flag shorthands.